chainer/reporter.py

```diff
@@ -356,3 +363,16 @@ def make_statistics(self):
             stats[name + '.std'] = std
 
         return stats
+
+    def serialize(self, serializer):
+        if isinstance(serializer, serializer_module.Serializer):
+            serializer('_names', json.dumps(list(self._summaries.keys())))
+            for name, summary in six.iteritems(self._summaries):
+                if hasattr(summary, 'serialize'):
+                    summary.serialize(serializer['_summaries'][name])
```

Comment:
Chainer's serializer uses `/` as a separator, so allowing arbitrary keys may cause error in edge cases.

```py
>>> import chainer
>>> s = chainer.DictSummary()
>>> s.add({'/a/b': 10, 'a/b': 20})
>>> chainer.serializers.save_npz('model.npz', s)
>>> s.compute_mean()
{'/a/b': 10.0, 'a/b': 20.0}
>>> chainer.serializers.load_npz('model.npz', s)
>>> s.compute_mean()
{'/a/b': 20.0, 'a/b': 20.0}
```

Reply:
Thank you for pointing out. I will use a hash (e.g. md5) for the key.

Reply:
Hm, I think using index is sufficient.

Reply:
I see.